userspace/libsinsp/dumper.cpp

```diff
@@ -47,13 +57,20 @@ void sinsp_dumper::open(const string& filename, bool compress, bool threads_from
 		m_inspector->m_thread_manager->to_scap();
 	}
 
-	if(compress)
+	if(m_target_memory_buffer)
```

Comment:
Hmm, I certainly could, but in such a simple scenario I don't see a big advantage in duplicating state and I would privilege simplicity over expressiveness.